packages/astro/src/vite-plugin-astro-server/pipeline.ts

```diff
@@ -133,6 +134,9 @@ export class DevPipeline extends Pipeline {
 
 	async preload(filePath: URL) {
 		const { loader } = this;
+		if (filePath.href === new URL('astro-default-404', this.config.root).href) {
+			return { default: default404Page } as any as ComponentInstance
```

Comment:
Should we use `unknown` instead of `any`? Or is the `as any` necessary?

Reply:
There wouldn't be any difference between the two in terms of safety.